cmd/install/addon/cmd.go

```diff
@@ -197,72 +196,53 @@ func run(cmd *cobra.Command, argv []string) {
 		parameters.Each(func(param *cmv1.AddOnParameter) bool {
 			var val string
 			var hasVal bool
+			var optionStrings []string
+
+			options, ok := param.GetOptions()
+			if ok {
+				for _, opt := range options {
+					optionStrings = append(optionStrings, opt.Value())
+				}
+			}
+
 			// If value is already set in the CLI, ignore interactive prompt
 			flag := cmd.Flags().Lookup(param.ID())
 			if flag != nil {
 				val = flag.Value.String()
 				hasVal = true
-			} else if interactive.Enabled() {
-				input := interactive.Input{
-					Question: param.Name(),
-					Help:     fmt.Sprintf("%s: %s", param.ID(), param.Description()),
-					Required: param.Required(),
-				}
-				// add a prompt to question name to indicate if the boolean param is required and check validation
-				if param.ValueType() == "boolean" && param.Validation() == "^true$" && param.Required() {
-					input.Question = fmt.Sprintf("%s (required)", param.Name())
-					input.Validators = []interactive.Validator{
-						interactive.RegExpBoolean(param.Validation()),
-					}
-				}
-				switch param.ValueType() {
-				case "boolean":
-					var boolVal bool
-					input.Default, _ = strconv.ParseBool(param.DefaultValue())
-					boolVal, err = interactive.GetBool(input)
-					if boolVal {
-						val = "true"
-					} else {
-						val = "false"
-					}
-				case "cidr":
-					var cidrVal net.IPNet
-					if param.DefaultValue() != "" {
-						_, defaultIDR, _ := net.ParseCIDR(param.DefaultValue())
-						input.Default = *defaultIDR
-					}
-					cidrVal, err = interactive.GetIPNet(input)
-					val = cidrVal.String()
-					if val == "<nil>" {
-						val = ""
-					}
-				case "number", "resource":
-					var numVal int
-					input.Default, _ = strconv.Atoi(param.DefaultValue())
-					numVal, err = interactive.GetInt(input)
-					val = fmt.Sprintf("%d", numVal)
-				case "string":
-					input.Default = param.DefaultValue()
-					val, err = interactive.GetString(input)
-				}
+			}
+			if interactive.Enabled() {
+				input := interactive.Input{}
+				input.Question = param.Name()
+				input.Help = fmt.Sprintf("%s: %s", param.ID(), param.Description())
+				input.Required = param.Required()
+				input.Options = optionStrings
+
+				val, err = interactive.GetAddonParameter(nil, param, input)
 				if err != nil {
-					r.Reporter.Errorf("Expected a valid value for '%s': %v", param.Name(), err)
+					r.Reporter.Errorf("%s", err)
 					os.Exit(1)
 				}
+
 				hasVal = true
 			}
+			if !hasVal {
+				return false
```

Comment:
This will cause the parameter loop to stop iterating, i.e. if one parameter is missing a value, the rest of the parameters will be skipped. With `.Each` loops, `return false` is equal to `break` and `return true` to `continue`.

tbh i'm not sure we really need this `hasVal` flag, but that's outside the scope of this PR.

Reply:
If I remove this return now that the logic is refactored, there isn't any use of the `hasVal` flag. I think it could be removed and the value of `val set to `""` on instantiation. I'm happy to do this in this pr if you agree?

Reply:
yes, please remove `hasVal` as it is not required any more